curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -213,16 +213,14 @@ private synchronized void checkTimeouts() throws Exception
      * Return the current session id
      */
     public long getSessionId() {
-        long sessionId = -1;
-        if (isConnected()) {
-            try {
-                ZooKeeper zk = zooKeeper.getZooKeeper();
-                if (zk != null) {
-                    sessionId = zk.getSessionId();
-                }
-            } catch (Exception e) {
-                // Ignore the exception
+        long sessionId = 0;
```

Comment:
My concern is that `zooKeeper.getZooKeeper()` can cause a new connection to get started. I'm not sure that that kind of side-effect is expected. If there is no current connection or the connection is in error somehow, `zooKeeper.getZooKeeper()` will call `HandleHolder.closeAndReset()`.